### PR TITLE
Add user configuration

### PR DESCRIPTION
**Pull request by Nusiq**
First commit added basic user_config reader:
- `project.app_data_storage` - replaces the useAppData property form the config.json file
- `user.name` - replaces the name property in config file created by 'regolith init'
- `resolvers` - is a list of URLs to 'resolver.json' files. This property is only partially implemented. Currently it only uses the first item from the list or the default value if the 'config.toml' doesn't specify any.

### CODE REVIEW

**Comment by Nusiq** [comment]
I think the help message is a nice summary of what the config command currently does:
```
The config command is used to manage the user configuration of Regolith. It can access and modify
global and project-specific configuration files. The global configuration file is stored in the
application data folder in the "user_config.json" file. The project-specific configuration file is
stored in the ".regolith/user_config.json". The project specific configuration file is used when
the local doesn't have specific properties defined. If the global configuration doesn't define them
either, the default values are used.

The behavior of the command changes based on the used flags and the number of provided arguments.
The cheetsheet below shows the possible combinations of flags and arguments and what they do:

Printing all properties:                        regolith config
Printing specified property:                    regolith config <key>
Setting property value:                         regolith config <key> <value>
Deleting a property:                            regolith config <key> --delete
Appending to a list proeprty:                   regolith config <key> <value> --append
Replacing item in a list property:              regolith config <key> <value> --index <index>
Deleting item in a list property:               regolith config <key> --index <index> --delete

You can use the "--global" or "--local" flags for all of the above combinations. To change whether
the command uses the global or local (project-specific) configuration file.

If you skip the "--global" or "--local" flag:
-  The commands that edit the config file use the local config file by default.
-  The commands that print the config file will print a combined view of the local and global
   configuration. This view defines, what Regolith will actually use when running the project.

You can use "regolith config" command to see the full configuration to learn what properties are
available.
```

File: go.mod
Comment (Nusiq) [inline go.mod:18]: We're using the unofficial version of go-getter here. We have to wait until this PR will be merged:

- https://github.com/hashicorp/go-getter/pull/355

**Comment by Nusiq** [comment]
I removed the `--local` flag and the option of saving the user config locally. It didn't make much sense to save user config in the `.regolith` path.

```
The config command is used to manage the user configuration of Regolith. It can access and modify
the user configuration file. The data is stored in the application data folder in the
"user_config.json" file.

The behavior of the command changes based on the used flags and the number of provided arguments.
The cheetsheet below shows the possible combinations of flags and arguments and what they do:

Printing all properties:                        regolith config
Printing specified property:                    regolith config <key>
Setting property value:                         regolith config <key> <value>
Deleting a property:                            regolith config <key> --delete
Appending to a list proeprty:                   regolith config <key> <value> --append
Replacing item in a list property:              regolith config <key> <value> --index <index>
Deleting item in a list property:               regolith config <key> --index <index> --delete

The printing commands can take the --full flag to print configuration with the default values
included (if they're not defined in the config file). Without the flag, the undefined properties
will be printed as null or empty list.

Usage:
  regolith config [key] [value] [flags]

Flags:
  -a, --append      Append value to array property
      --debug       Enables debugging
  -d, --delete      Delete property
  -f, --full        When printing, prints the full configuration including default values.
  -h, --help        help for config
  -i, --index int   The index of the array property on which to act (default -1)
  ```

**Comment by Nusiq** [comment]
This PR closes #225

**Review by stirante — APPROVED** [review]

**Pull request merged**
